### PR TITLE
Added support for minimumversion, maximumversion and requiredversion

### PR DESCRIPTION
**Pull request by gustavdw**

### CODE REVIEW

**Comment by gustavdw** [comment]
I've added the installscope as well.

File: PowerShellModule/PowerShellModule.psm1 [hunk 1]
```diff
@@ -25,58 +41,122 @@ class PSModuleResource {
...
             catch {
                 Write-Error -ErrorRecord $_
             }
         }
         elseif ($this.Ensure -eq 'absent') {
-            Uninstall-Module -Name $this.Module_Name -Force
+            $arguments = $this.GetVersionArguments()
+            $arguments.Add("-Name", $this.Module_Name)
+            $arguments.Add("-Force", $true)
+            Uninstall-Module @arguments
```
Comment (davidobrien1985) [inline PowerShellModule/PowerShellModule.psm1:83]: Why not write the arguments as a hashtable?

File: PowerShellModule/PowerShellModule.psm1 [hunk 2]
```diff
@@ -25,58 +41,122 @@ class PSModuleResource {
...
+    [bool] Test() {
 
-        if ($Module -and ($this.Ensure -eq 'present')) {
-            return [bool] $true
+        $modules = @()
+        $modules += @(Get-Module -Name $this.Module_Name -ListAvailable -ErrorAction Ignore)
+        $returnVal = $false
+
+        Write-Verbose "The following versions of $($this.Module_Name) are installed"
+        $modules | % {
```
Comment (davidobrien1985) [inline PowerShellModule/PowerShellModule.psm1]: Would rather see a `foreach` here

File: PowerShellModule/Readme.md [hunk 3]
```diff
@@ -24,6 +24,36 @@ Via Invoke-DscResource:
...
+    Ensure = 'present'
+    Module_Name = 'AzureExt'
+    MinimumVersion = '1.0.0.0'
+    MaximumVersion = '2.0.0.0'
+    InstallScope = 'allusers'        
+}
+````
 
 ## More Info:
 Read http://www.david-obrien.net/?p=3108
```
Comment (davidobrien1985) [inline PowerShellModule/Readme.md]: Forgot to update this myself...
Mind changing this to `https://david-obrien.net/2015/09/powershell-dsc-to-manage-powershell-modules/` ? :)

**Review by davidobrien1985 — COMMENTED** [review]
Just minor things.
Thanks for contributing!!!

File: PowerShellModule/PowerShellModule.psm1 (hunk 1 above)
Comment (gustavdw) [reply PowerShellModule/PowerShellModule.psm1:83]: Refactored.

File: PowerShellModule/PowerShellModule.psm1 (hunk 2 above)
Comment (gustavdw) [reply PowerShellModule/PowerShellModule.psm1]: Fixed!

File: PowerShellModule/Readme.md (hunk 3 above)
Comment (gustavdw) [reply PowerShellModule/Readme.md]: Yepp!

**Comment by gustavdw** [comment]
@nzspambot will sort out next week.

**Comment by gustavdw** [comment]
@nzspambot @davidobrien1985 
Added support for custom repositories (tested against a custom repository/gallery in ProGet with a connector to powershellgallery.com). It's implemented as a seperate resource which you can depend on from the psmoduleresource. Also updated the instructions/readme to new format with breakdown of resources and properties, with examples listed below. Set some default values in both resources for Ensure ('present').

Let me know if you have any feedback!

**Pull request merged**